python-package/pyproject.toml

```diff
@@ -1,3 +1,63 @@
+[project]
+
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
```

Comment:
Shouldn't Python 3.6 be here given that we have `requires-python = ">=3.6"`?

Reply:
oh sure, I can add that.

It's not there in `setup.py` and I was just copying from that.

https://github.com/microsoft/LightGBM/blob/a323c7583ccdffb7aff49e7f9ca72042aca18395/python-package/setup.py#L341-L355

Reply:
refer to https://github.com/microsoft/LightGBM/pull/4891

Reply:
Ah right, thanks @guolinke for reminding me ... in #4891 we dropped support, meaning:

* dropped that trove classifier `'Programming Language :: Python :: 3.6'` from the PyPI labels
* do not have any CI jobs testing Python 3.6

And we decided not to move up to `requires-python = ">=3.7"` so that people could *technically* try to build for Python 3.6 if they needed that a bit longer (as @jmoralez also referred to in https://github.com/microsoft/LightGBM/pull/5765#issuecomment-1454861675).

By the way, I tried tonight and it looks like `lightgbm` is still at least minimally compatible with Python 3.6 🎉 .

```shell
docker run \
    --rm \
    -it python:3.6 \
    bash

apt-get update
apt-get install \
    --no-install-recommends \
    -y \
        cmake

git clone --recursive \
    https://github.com/microsoft/LightGBM.git

cd LightGBM
sh ./build-python.sh install --nomp

pip install pandas
python ./examples/python-guide/simple_example.py
```

logs:

```text
Loading data...
Starting training...
[LightGBM] [Warning] Auto-choosing col-wise multi-threading, the overhead of testing was 0.002058 seconds.
You can set `force_col_wise=true` to remove the overhead.
Training until validation scores don't improve for 5 rounds
Did not meet early stopping. Best iteration is:
[20]	valid_0's l2: 0.198063	valid_0's l1: 0.431129
Saving model...
Starting predicting...
The RMSE of prediction is: 0.4450426449744025
```

I don't feel strongly either way about whether or not to have the Python 3.6 trove classifier. Since @StrikerRUS removed it in #4891, I think we should defer to his judgment and continue to omit it. Is that ok @jmoralez ? If, after reading this, you still feel strongly that it should be included since `lightgbm` at least *minimally* supports Python 3.6, I'll add it.

Reply:
That's ok, I agree with not adding it back.

Reply:
Alright, thanks for talking through it with us! In another PR, I might put up a minimal CI job that runs exactly that dockerized little script I just shared above. So we can rely on CI to catch issues of the form "this PR would be a hard compatibility break because it's not even syntactically valid Python 3.6".

I don't want to take on any more support than that for Python 3.6, but would be useful (I think) to get automated feedback at that level at least, and I'd expect that job shouldn't require too much maintenance burden.

Reply:
For those finding this conversation from search... here's the PR I promised: #5936.